### PR TITLE
Refactoring config file to account for common config

### LINKED ISSUE #655: Use spk-config in azurecredentials
**As a:** developer

**I want:** to use the spk-config file

**So that:** azurecredentials doesn't have to rely on the config.ts file

**Describe the solution you'd like:** delete config.ts and refactor azurecredentials.ts to use values from `spk-config.yaml`

**Does this require updates to documentation?:** no

### PR DESCRIPTION
**Pull request by samiyaakhtar**
Moved some things to common areas, which belong under specific AzDO or AzDO pipelines sections. 

Closes https://github.com/microsoft/bedrock/issues/655

### CODE REVIEW

File: spk-config.yaml [hunk 1]
```diff
@@ -1,45 +1,52 @@
...
+  tenant_id: "tenant-id"
+
+# Azure CLI Specific Configuration
+azure-cli:
+  version: "2.0.70"
+  Extensions:
+    aks-preview: "0.4.14"
+
+# Azure DevOps Configuration
+azure-devops: # Should this live under the `spk` configuration?
```
Comment (bnookala) [inline spk-config.yaml]: I think we may need to enumerate the other repositories here as well (ie hld, materialized manifest)

**Review by bnookala — CHANGES_REQUESTED** [review]

File: spk-config.yaml (hunk 1 above)
Comment (samiyaakhtar) [reply spk-config.yaml]: You mean for the case when hld, manifest may be under a different azure org and project? cc: @mtarng

Comment (bnookala) [reply spk-config.yaml]: no, we just need it for creating pipelines between them - let's assume one project for now

**Review by yradsmikham — APPROVED** [review]

**Review by edaena — APPROVED** [review]

File: spk-config.yaml [hunk 2]
```diff
@@ -1,45 +1,35 @@
+# Azure CLI Specific Configuration
+azure_cli:
+  version: "2.0.70"
+  Extensions:
+    aks_preview: "0.4.14"
+
+# Azure DevOps Configuration
+azure_devops: # Should this live under the `spk` configuration?
+  org: "my-az-do-org"
+  project: "my-mono-repo-project"
```
Comment (bnookala) [inline spk-config.yaml:10]: i think you can remove this, since it's in a separate config

**Review by bnookala — APPROVED** [review]

**Pull request merged**